File: BikeDemoJava/app/src/main/java/com/magiclane/sdk/examples/bikedemojava/SettingsSwitchItem.java
/*
 * SPDX-FileCopyrightText: 2026 Magic Lane International B.V. <info@magiclane.com>
 * SPDX-License-Identifier: Apache-2.0
 *
 * Contact Magic Lane at <info@magiclane.com> for SDK licensing options.
 */

package com.magiclane.sdk.examples.bikedemojava;

public class SettingsSwitchItem extends SettingsItem {
    private final boolean itIs;
    private final SwitchCallback callback;

    public interface SwitchCallback {
        void onChanged(boolean value);
    }

    public SettingsSwitchItem(String title, boolean itIs, SwitchCallback callback) {
        super(title);
        this.itIs = itIs;
        this.callback = callback;
    }

    public boolean isItIs() {
        return itIs;
    }

    public SwitchCallback getCallback() {
        return callback;
    }
}

File: BikeSimulation/app/src/main/kotlin/com/magiclane/sdk/examples/bikesimulation/SettingsSwitchItem.kt
// -------------------------------------------------------------------------------------------------

/*
 * SPDX-FileCopyrightText: 1995-2025 Magic Lane International B.V. <info@magiclane.com>
 * SPDX-License-Identifier: Apache-2.0
 *
 * Contact Magic Lane at <info@magiclane.com> for SDK licensing options.
 */

// -------------------------------------------------------------------------------------------------

package com.magiclane.sdk.examples.bikesimulation

// -------------------------------------------------------------------------------------------------

data class SettingsSwitchItem(
    override val title: String = "",
    val itIs : Boolean = false,
    val callback: (Boolean) -> Unit
):SettingsItem(title)
// -------------------------------------------------------------------------------------------------
